src/ucp/core/ucp_worker.c

```diff
@@ -1569,7 +1574,30 @@ ucs_status_t ucp_worker_query(ucp_worker_h worker,
         }
     }
 
-    return UCS_OK;
+    if (attr->field_mask & UCP_WORKER_ATTR_FIELD_ADDRESS) {
+        /* If UCP_WORKER_ATTR_FIELD_ADDRESS_FLAGS is not set or
+           address flags is 0, pack all tl adresses.  */
+        tl_bitmap = -1;
+
+        if (attr->field_mask & UCP_WORKER_ATTR_FIELD_ADDRESS_FLAGS) {
+            if (attr->address_flags == UCP_WORKER_FLAG_NET_ADDRESS) {
+                tl_bitmap = 0;
+                ucs_for_each_bit(tl_id, context->tl_bitmap) {
+                    if (context->tl_rscs[tl_id].tl_rsc.dev_type == UCT_DEVICE_TYPE_NET) {
+                        tl_bitmap |= UCS_BIT(tl_id);
+                    }
+                }
+            } else if (attr->address_flags != 0) {
```

Comment:
looks like this check is not future-compatible. why not just ignore unknown flags?